server/worker/worker.go

```diff
@@ -191,6 +209,7 @@ func (w *Worker) processJob(ctx context.Context, job *fleet.Job) error {
 		args = *job.Args
 	}
 
+	ctx = context.WithValue(ctx, retryNumberCtxKey, job.Retries)
```

Comment:
A bit hacky, but that was a quick and easy way to make that info available to the job processor.